### PR TITLE
updating numpy + vtk functions

### PR DESCRIPTION
**Pull request by AurelienBesnier**
Hi !
I updated some of the imports of vtk modules which were outdated and also made some updates of numpy functions to be compatible with 2.0 and 1.x.

For the compatibility with numpy 2.0 we still need to wait for the skan package to be updated.

Also with @christian34 we investigated the problems we had with the CI and they are now fixed.

### CODE REVIEW

**Comment by pradal** [comment]
Excellent.
You introduce a new dependency : vtkmodule.
Is it provided by vtk?
Otherelse you have to add it to meta.yaml

**Review by pradal — APPROVED** [review]
This sound great!
Thank you for modernize the code
Just remove cython if not used anymore

File: conda/meta.yaml [hunk 1]
```diff
@@ -10,12 +10,12 @@ build:
...
-    - {{PYTHON}} setup.py build_ext -j{{ CPU_COUNT }}
     - {{ PYTHON }} -m pip install . -vv
 
 requirements:
   host:
     - python
+    - setuptools
     - numpy 
     - scipy
     - cython
```
[the remark this answers is not in the document]
Comment (christian34) [reply conda/meta.yaml:21]: I do not think that we want to remove cython, as we want the two c extensions for acceleration !

[the remark this answers is not in the document]
Comment (pradal) [reply conda/meta.yaml:21]: I misunderstood some commit. I was a little bit afraid...

**Comment by AurelienBesnier** [comment]
The vtkmodules is included in the vtk package. As per the [vtk documentation](https://github.com/Kitware/VTK/blob/master/Wrapping/Python/README.md):

The 'vtk.py' module is obsolete and is only provided for backwards compatibility. When you import vtk, this module actually provides you with the 'vtkmodules.all' module via some internal trickery:
```
>>> import vtk
>>> vtk.__name__
'vtkmodules.all'
```

**Pull request merged**